addons/mail/static/src/js/utils.js

```diff
@@ -53,10 +53,17 @@ function linkify(text, attrs) {
     attrs = _.map(attrs, function (value, key) {
         return key + '="' + _.escape(value) + '"';
     }).join(' ');
-    return text.replace(urlRegexp, function (url) {
+    var curIndex = 0;
+    var result = "";
+    var match;
+    while ((match = urlRegexp.exec(text)) !== null) {
+        result += _.escape(text.slice(curIndex, match.index));
+        var url = _.escape(match[0]);
         var href = (!/^https?:\/\//i.test(url)) ? "http://" + url : url;
-        return '<a ' + attrs + ' href="' + href + '">' + url + '</a>';
-    });
+        result += '<a ' + attrs + ' href="' + href + '">' + url + '</a>';
+        curIndex = match.index + match[0].length;
+    }
+    return result + _.escape(text.slice(curIndex));
```

Comment:
Bad escape function as explained in the test comment, the one from OWL would resolve all issues:

```
function escape(str) {
  if (str === undefined) {
    return "";
  }
  if (typeof str === "number") {
    return String(str);
  }
  const p = document.createElement("p");
  p.textContent = str;
  return p.innerHTML;
}

```